From 2868aace126b0fc2cbff56337e02b573d9302b4b Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Thu, 28 Jan 2021 06:53:12 +0200
Subject: [PATCH] [SW-20154] make hash size platform specific

Goya2 uses SHA256 and Gaudi2 uses SHA384. Since the
hash sizes are per platform, move them to platform
specific folder.

Change-Id: I86fffbcdc9524121bbc656b88c147c13f761fd6e
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/security.h b/common/security.h
index 819adec..a2de5e3 100644
--- a/common/security.h
+++ b/common/security.h
@@ -9,6 +9,8 @@
 #define _SECURITY_H_
 #include <limits.h>
 
+#include <dev_security.h>
+
 enum security_img_status {
 	IMG_STATUS_FW_VERSION_LOW,
 	IMG_STATUS_SIG_AUTH_FAIL,
@@ -33,7 +35,6 @@
 
 #define SECURITY_KEY_HASH_COUNT				5
 #define SECURITY_KEY_HASH_HABANA_COUNT			1
-#define SECURITY_DWORD_PER_KEY				8
 #define SECURITY_DWORD_BIT_SIZE				(sizeof(uint32_t) * \
 							 CHAR_BIT)
 #define SECURITY_BYTE_PER_DWORD				(sizeof(uint32_t))
diff --git a/gaudi2_arc/dev_security.h b/gaudi2_arc/dev_security.h
new file mode 100644
index 0000000..ad70e78
--- /dev/null
+++ b/gaudi2_arc/dev_security.h
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+#ifndef _GAUDI2_DEV_SECURITY_H_
+#define _GAUDI2_DEV_SECURITY_H_
+
+#define SECURITY_DWORD_PER_KEY				12
+
+#endif /* _GAUDI2_DEV_SECURITY_H_ */
diff --git a/goya2_arc/dev_security.h b/goya2_arc/dev_security.h
new file mode 100644
index 0000000..1d812e9
--- /dev/null
+++ b/goya2_arc/dev_security.h
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ * All Rights Reserved.
+ *
+ */
+
+#ifndef _GOYA2_DEV_SECURITY_H_
+#define _GOYA2_DEV_SECURITY_H_
+
+#define SECURITY_DWORD_PER_KEY				8
+
+#endif /* _GOYA2_DEV_SECURITY_H_ */
